staging/src/k8s.io/client-go/tools/clientcmd/api/v1/conversion.go

```diff
@@ -81,6 +81,7 @@ func Convert_Slice_v1_NamedAuthInfo_To_Map_string_To_Pointer_api_AuthInfo(in *[]
 }
 
 func Convert_Map_string_To_Pointer_api_AuthInfo_To_Slice_v1_NamedAuthInfo(in *map[string]*api.AuthInfo, out *[]NamedAuthInfo, s conversion.Scope) error {
+	*out = make([]NamedAuthInfo, 0, len(*in))
```

Comment:
for reviewer: this line is the major modfication

Reply:
What about 
```
if *out == nil {
    *out = make([]NamedAuthInfo, 0, len(*in))
}
```

Reply:
this function is expected to convert a map of `AuthInfo` to a list, I think it might not be necessary to check `*out` here. 

In addition, suppose user passes a non-empty list as `out` and we do not reset it here, the `AuthInfo`s in the `in` is appended to `*out`, so `*out` may be different from `*in` afterwards. I guess this is not intended.

Reply:
This will make this case exceptional, look at others, like contexts, for example. There empty slice also gives you null and that's fine. Read more here https://stackoverflow.com/questions/33183071/golang-serialize-deserialize-an-empty-array-not-as-null about why it is this way.